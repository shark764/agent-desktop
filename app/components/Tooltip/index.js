/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/**
 *
 * Tooltip
 *
 */

/**
 *
 * Example:
 *
 * <Button
 *   data-tip
 *   data-for={tooltipTextObj.id}
 * />
 *
 * <Tooltip text={tooltipTextObj} id={tooltipTextObj.id}/>
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import Radium from 'radium';

import ReactTooltip from 'react-tooltip';

const styles = {
  defaultTheme: {
    color: '#ff6e00 !important',
    backgroundColor: 'orange !important',
  },
};

function Tooltip(props) {
  return (
    <ReactTooltip
      id={props.id}
      type={props.type}
      effect={props.effect}
      place={props.place}
      style={styles.defaultTheme}
    >
      <FormattedMessage {...props.text} />
    </ReactTooltip>
  );
}

Tooltip.propTypes = {
  id: PropTypes.string,
  text: PropTypes.shape({
    id: PropTypes.string,
    defaultMessage: PropTypes.string,
  }).isRequired,
  type: PropTypes.oneOf(['success', 'warning', 'error', 'info', 'light']),
  effect: PropTypes.oneOf(['float', 'solid']),
  place: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
};

Tooltip.defaultProps = {
  type: 'error',
  effect: 'solid',
  place: 'bottom',
};

export default Radium(Tooltip);
