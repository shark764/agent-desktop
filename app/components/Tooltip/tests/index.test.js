/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import Tooltip from '../index';

describe('<Tooltip />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <div id="mockWrapper" data-tip data-for="mockId">
        <Tooltip
          id="mockId"
          text={{ id: 'mockId', defaultMessage: 'mockMsg' }}
        />
      </div>
    );
    expect(rendered).toMatchSnapshot();
  });
});
