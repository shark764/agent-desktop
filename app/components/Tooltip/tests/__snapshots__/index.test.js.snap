// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render correctly 1`] = `
<div
  data-for="mockId"
  data-tip={true}
  id="mockWrapper"
>
  <Tooltip
    effect="solid"
    id="mockId"
    place="bottom"
    text={
      Object {
        "defaultMessage": "mockMsg",
        "id": "mockId",
      }
    }
    type="error"
  />
</div>
`;
